sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -315,9 +315,28 @@ object GpuDeviceManager extends Logging {
       }
 
       Cuda.setDevice(gpuId)
-      Rmm.initialize(init, logConf, poolAllocation)
-      RapidsBufferCatalog.init(conf)
+      try {
+        Rmm.initialize(init, logConf, poolAllocation)
+      } catch {
+        case firstEx: CudfException if ((init | RmmAllocationMode.CUDA_ASYNC) != 0) => {
+          logWarning("Failed to initialize RMM with ASYNC allocator. " +
+            "Try to initialize it again with ARENA allocator as a fallback option.")
+          init = init & (~RmmAllocationMode.CUDA_ASYNC) & RmmAllocationMode.ARENA
+          try {
+            Rmm.initialize(init, logConf, poolAllocation)
+          } catch {
+            case secondEx: Throwable => {
+              logWarning("Failed to initialize RMM again with ARENA allocator. " +
```

Comment:
Please change the level to `logError`. 

I would change the message to "Failed to initialize RMM with either the ASYNC or ARENA allocators. Exiting!"

Reply:
Will `logError` terminate the application before throwing the exception in the following line?

Reply:
no, it's just a different logging level